sway-core/src/control_flow_analysis/dead_code_analysis.rs

```diff
@@ -1729,3 +1778,116 @@ fn connect_storage_declaration<'eng: 'cfg, 'cfg>(
 
     graph.namespace.insert_storage(field_nodes);
 }
+
+/// Checks [AttributesMap] for `#[allow(dead_code)]` usage, if so returns true
+/// otherwise returns false.
+fn allow_dead_code(attributes: AttributesMap) -> bool {
+    fn allow_dead_code_helper(attributes: AttributesMap) -> Option<bool> {
+        if attributes
+            .get(&transform::AttributeKind::Allow)?
+            .last()?
+            .args
+            .first()?
+            .as_str()
+            == ALLOW_DEAD_CODE_NAME
+        {
+            return Some(true);
```

Comment:
I used the helper method that returns an option because the helper uses `?` which told me
```
the `?` operator can only be used in a function that returns `Result` or `Option`
```